src/store/router.saga.js

```diff
@@ -13,13 +13,23 @@ export function* locationChangeSaga({
 }) {
     const detailsMatch = matchPath(pathname, routes.DETAILS);
     const homeMatch = matchPath(pathname, { path: routes.HOME, exact: true });
+    const searchMatch = matchPath(pathname, routes.SEARCH);
 
     if (detailsMatch) {
```

Comment:
while the app will grow this method will become larger. I suggest you to split this logic. Need to be discussed.